src/services/ChallengeService.js

```diff
@@ -921,6 +921,30 @@ searchChallenges.schema = {
 async function createChallenge(currentUser, challenge, userToken) {
   await challengeHelper.validateCreateChallengeRequest(currentUser, challenge);
 
+  //Validate the groups if Valid or Not
+  if(challenge.groups)
+  {
+      for( const group of challenge.groups)
+      {
+        let res;
+        try
+        {
+            res = await axios.get(`https://api.topcoder-dev.com/v5/groups/${group}`,
+                  {
+                      headers: { Authorization: `Bearer ${token}` },
+                  });
+        }
+        catch(error)
+        {
+          throw new errors.BadRequestError(`The selected group ${group} does not exist as GET returned status ${error.response.status}`);
+        }
+        if(res.data.status==='inactive')
+        {
+          throw new errors.BadRequestError(`The selected group ${group} is inactive`);
+        }
+      }
+  }
+
```

Comment:
Let's remove all newly added code.
We don't need to check groups one by one.
There is already a group validation method in both challenge create and update service.
It just performs the validation if user is not admin or m2m.
What needs to be done is to expand that validation methods in both create/update to do:
1. If user is not admin or m2m, do the same thing we do today. It is fetching the member groups and compares to see if member has access to that group. With this way we already validate if group existst. But we need to include checking the group status here.
2. If user is admin or m2m check only if group exists, as you do it here. But follow the instructions provided by @liuliquan

Reply:
deleteChallenge uses ensureUserCanModifyChallenge.
So it may introduce additional bugs as